packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -108,57 +109,58 @@ export class BlobManager {
         // When this blob is retrieved, let the container runtime know that the corresponding GC node got updated.
         this.gcNodeUpdated(this.getBlobGCNodePath(blobId));
 
-        return new BlobHandle(
-            `${BlobManager.basePath}/${storageId}`,
-            this.routeContext,
+        return PerformanceEvent.timedExecAsync(
+            this.logger,
+            { eventName: "AttachmentReadBlob", id: storageId },
             async () => {
-                return this.getStorage().readBlob(storageId).catch((error) => {
-                    this.logger.sendErrorEvent(
-                        {
-                            eventName: "AttachmentReadBlobError",
-                            id: storageId,
-                        },
-                        error,
-                    );
-                    throw error;
-                });
+                return new BlobHandle(
+                    `${BlobManager.basePath}/${storageId}`,
+                    this.routeContext,
+                    async () => this.getStorage().readBlob(storageId),
+                );
             },
-        );
+            { start: true, end: true, cancel: "error" });
```

Comment:
i also added the start events, but on second thought do we need them?

Reply:
i;m looking if cancel should be generic or error

Reply:
I'd not record start event - I doubt we will ever need it. _end & _cancel are enough